packages/bierzo-wallet/src/communication/signAndPost/index.ts

```diff
@@ -1,52 +1,66 @@
 /*global chrome*/
 import {
   Address,
+  Amount,
+  ChainId,
   Identity,
   SendTransaction,
-  TokenTicker,
   TransactionId,
   UnsignedTransaction,
   WithCreator,
 } from '@iov/bcp';
 import { TransactionEncoder } from '@iov/core';
-import { ethereumCodec, EthereumConnection } from '@iov/ethereum';
 import { isJsonRpcErrorResponse, JsonRpcRequest, makeJsonRpcId, parseJsonRpcResponse2 } from '@iov/jsonrpc';
 
 import { extensionId } from '..';
+import { getCodecForChainId } from '../../logic/codec';
+import { getConnectionForChainId } from '../../logic/connection';
 
-async function withEthereumFee<T extends UnsignedTransaction>(transaction: T): Promise<T> {
-  const connection = await EthereumConnection.establish('http://localhost:8545', {});
+async function withChainFee<T extends UnsignedTransaction>(chainId: ChainId, transaction: T): Promise<T> {
+  const connection = await getConnectionForChainId(chainId);
   const withFee = await connection.withDefaultFee(transaction);
-  connection.disconnect();
   return withFee;
 }
 
-const generateSignAndPostRequest = async (creator: Identity): Promise<JsonRpcRequest> => {
-  const transactionWithFee: SendTransaction & WithCreator = await withEthereumFee({
+const generateSignAndPostRequest = async (
+  chainId: ChainId,
+  creator: Identity,
+  recipient: Address,
+  amount: Amount,
+  memo: string | undefined,
+): Promise<JsonRpcRequest> => {
+  const codec = await getCodecForChainId(chainId);
+
+  const transactionWithFee: SendTransaction & WithCreator = await withChainFee(chainId, {
     kind: 'bcp/send',
-    recipient: '0x0000000000000000000000000000000000000000' as Address,
-    creator: creator,
-    sender: ethereumCodec.identityToAddress(creator),
-    amount: {
-      quantity: '1234000000000000000',
-      fractionalDigits: 18,
-      tokenTicker: 'ETH' as TokenTicker,
-    },
+    recipient,
+    creator,
+    sender: codec.identityToAddress(creator),
+    amount: amount,
+    memo: memo ? memo : '',
```

Comment:
Why this case distinction? You can pass `string | undefined` into `memo` directly

Reply:
Unfortunately not :( I got 'Cannot encode type to JSON' if I tried to use undefined `memo`. It happens during [this call](https://github.com/iov-one/ponferrada/blob/a961be138e79875158391ed779da55988c9c82f3/packages/bierzo-wallet/src/communication/signAndPost/index.ts#L42).

Reply:
Ah okay, this is a problem that needs fixing. Thanks for letting know. I will find a solution on the IOV-Core side. For now, your workaround is fine.